src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -47,66 +47,102 @@ private static void FreeDeconstructionVariables(ArrayBuilder<DeconstructionVaria
         /// Deconstruct steps for tuples have no invocation to Deconstruct, but steps for non-tuples do.
         /// The caller is responsible for releasing all the ArrayBuilders in checkedVariables.
         /// </summary>
-        private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CSharpSyntaxNode node, ExpressionSyntax right, ArrayBuilder<DeconstructionVariable> checkedVariables, DiagnosticBag diagnostics, bool isDeclaration, BoundDeconstructValuePlaceholder rhsPlaceholder = null)
+        private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(
+                                                        CSharpSyntaxNode node,
+                                                        ExpressionSyntax right,
+                                                        ArrayBuilder<DeconstructionVariable> checkedVariables,
+                                                        DiagnosticBag diagnostics,
+                                                        bool isDeclaration,
+                                                        BoundDeconstructValuePlaceholder rhsPlaceholder = null)
         {
-            TypeSymbol voidType = GetSpecialType(SpecialType.System_Void, diagnostics, node);
-
             // receiver for first Deconstruct step
             var boundRHS = rhsPlaceholder ?? BindValue(right, diagnostics, BindValueKind.RValue);
 
+            boundRHS = FixTupleLiteral(checkedVariables, boundRHS, node, right, diagnostics);
```

Comment:
So the right-hand-side is already given the final type?

---
In reply to: [74643437](https://github.com/dotnet/roslyn/pull/13115#discussion_r74643437) [](ancestors = 74643437)